### PR TITLE
Introduce a function for date range events

### PR DESCRIPTION
**Pull request by pantierra**

### CODE REVIEW

File: pkg/calendar/calendar.go [hunk 1]
```diff
@@ -42,7 +42,21 @@ func (cal *Calendar) SetTimezone(tz *time.Location) {
...
 }
 
-func (cal Calendar) GetEventsOn(date time.Time) ([]ical.Event, error) {
+func (cal Calendar) GetEventsOfRange(start time.Time, end time.Time) ([]ical.Event, error) {
+	var allEvents []ical.Event
+
+	for d := start; !d.After(end); d = d.AddDate(0, 0, 1) {
+		dailyEvents, err := cal.GetEventsOnDay(d)
+		if err != nil {
+			return nil, err
```
Comment (gueldenstone) [inline pkg/calendar/calendar.go]: I would rather just skip the day then if there is an error for a specific day. I'd just `continue` if there is an error and maybe log it.

File: pkg/calendar/calendar.go [hunk 2]
```diff
@@ -42,7 +42,21 @@ func (cal *Calendar) SetTimezone(tz *time.Location) {
 	cal.tz = tz
 }
 
-func (cal Calendar) GetEventsOn(date time.Time) ([]ical.Event, error) {
+func (cal Calendar) GetEventsOfRange(start time.Time, end time.Time) ([]ical.Event, error) {
+	var allEvents []ical.Event
```
Comment (gueldenstone) [inline pkg/calendar/calendar.go]: Please consider initializing this variable for the reasons I mentioned in #6.

**Review by gueldenstone — COMMENTED** [review]
Would be great to have some tests even if they are based on the exisiting small set of test data.

File: pkg/calendar/calendar.go (hunk 1 above)
Comment (pantierra) [reply pkg/calendar/calendar.go]: Adjusted.

File: pkg/calendar/calendar.go (hunk 2 above)
Comment (pantierra) [reply pkg/calendar/calendar.go]: I think there is no `nil` return anymore.

**Pull request closed without merging**